pkg/security/ptracer/fim_handlers.go

```diff
@@ -1193,14 +1206,34 @@ func handleFcntlRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg,
 
 func handleDupRet(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
 	if ret := tracer.ReadRet(regs); msg.Dup != nil && ret >= 0 {
-		if path, ok := process.FdRes.Fd[msg.Dup.OldFd]; ok {
+		if path, err := process.GetFilenameFromFd(msg.Dup.OldFd); err == nil {
 			// maintain fd/path in case of dups
 			process.FdRes.Fd[int32(ret)] = path
 		}
 	}
 	return nil
 }
 
+const sizeOfInt = 4
+
+func handlePipe2Ret(tracer *Tracer, process *Process, msg *ebpfless.SyscallMsg, regs syscall.PtraceRegs, _ bool) error {
+	fds, err := tracer.readData(process.Pid, msg.Pipe.FdsPtr, 2*sizeOfInt)
```

Comment:
We should start by checking the ret value here I guess, and exit if an error was returned